plugins/utils/mkdnflow.nix

```diff
@@ -0,0 +1,612 @@
+              Set to `false` if you wish to set mappings outside of the plugin.
+            '';
+          };
+          paths = {
+            default = true;
+            description = "Required for link interpretation and following links.";
+          };
+          tables = {
+            default = true;
+            description = "Required for table navigation and formatting.";
+          };
+          yaml = {
+            default = false;
+            description = "Required for parsing yaml blocks.";
+          };
+        };
+
+      filetypes =
+        helpers.defaultNullOpts.mkNullable
+        (with types; attrsOf bool)
+        "{md = true; rmd = true; markdown = true;}"
+        ''
+          A matching extension will enable the plugin's functionality for a file with that
+          extension.
+
+          NOTE: This functionality references the file's extension. It does not rely on
+          Neovim's filetype recognition.
+          The extension must be provided in lower case because the plugin converts file names to
+          lowercase.
+          Any arbitrary extension can be supplied.
+          Setting an extension to `false` is the same as not including it in the list.
+        '';
+
+      createDirs = helpers.defaultNullOpts.mkBool true ''
+        - `true`: Directories referenced in a link will be (recursively) created if they do not
+          exist.
+        - `false`: No action will be taken when directories referenced in a link do not exist.
+          Neovim will open a new file, but you will get an error when you attempt to write the
+          file.
+      '';
+
+      perspective = {
+        priority = helpers.defaultNullOpts.mkEnumFirstDefault ["first" "current" "root"] ''
+          Specifies the priority perspective to take when interpreting link paths
+
+          - `first` (default): Links will be interpreted relative to the first-opened file (when
+            the current instance of Neovim was started)
+          - `current`: Links will be interpreted relative to the current file
+          - `root`: Links will be interpreted relative to the root directory of the current
+            notebook (requires `perspective.root_tell` to be specified)
+        '';
+
+        fallback = helpers.defaultNullOpts.mkEnumFirstDefault ["first" "current" "root"] ''
+          Specifies the backup perspective to take if priority isn't possible (e.g. if it is
+          `root` but no root directory is found).
+
+          - `first` (default): Links will be interpreted relative to the first-opened file (when
+            the current instance of Neovim was started)
+          - `current`: Links will be interpreted relative to the current file
+          - `root`: Links will be interpreted relative to the root directory of the current
+            notebook (requires `perspective.root_tell` to be specified)
+        '';
+
+        rootTell = helpers.defaultNullOpts.mkNullable (with types; either bool str) "false" ''
```

Comment:
maybe either str or (enum [false]) as is done later

Reply:
Done